src/dao.ts

```diff
@@ -170,59 +163,33 @@ export class DAO implements IStateful<IDAOState> {
       }
       ${DAO.fragments.DAOFields}
      `
-
-    const itemMap = (item: any): IDAOState => {
-      if (item === null) {
-        throw Error(`Could not find a DAO with id ${this.id}`)
-      }
-      const reputation = new Reputation(this.context, item.nativeReputation.id)
-      const token = new Token(this.context, item.nativeToken.id)
-      const state = {
-        address: item.id,
-        dao: this,
-        id: item.id,
-        memberCount: Number(item.reputationHoldersCount),
-        name: item.name,
-        numberOfBoostedProposals: Number(item.numberOfBoostedProposals),
-        numberOfPreBoostedProposals: Number(item.numberOfPreBoostedProposals),
-        numberOfQueuedProposals: Number(item.numberOfQueuedProposals),
-        register: item.register,
-        reputation,
-        reputationTotalSupply: new BN(item.nativeReputation.totalSupply),
-        token,
-        tokenName: item.nativeToken.name,
-        tokenSymbol: item.nativeToken.symbol,
-        tokenTotalSupply: item.nativeToken.totalSupply
-      }
-      this.setState(state)
-      return state
-    }
-    return this.context.getObservableObject(query, itemMap, apolloQueryOptions)
+     
+    return this.context.getObservableObject(this.context, query, DAO.itemMap, apolloQueryOptions)
   }
 
-  /*
-   * return the nativeReputation of the DAO
-   * @returns an (Observable) that returns a Reputation instance
-   */
   public nativeReputation(): Observable<Reputation> {
-    return this.state().pipe(first()).pipe(map((r) => r.reputation))
+    return this.state().pipe(first()).pipe(map((r) => r.reputation.entity))
+  }
+
+  public ethBalance(): Observable<BN> {
+    return this.context.ethBalance(this.id)
   }
 
-  public schemes(
-    options: ISchemeQueryOptions = {},
+  public plugins(
+    options: IPluginQueryOptions = {},
     apolloQueryOptions: IApolloQueryOptions = {}
-  ): Observable<SchemeBase[]> {
+  ): Observable<AnyProposalPlugin[]> {
     if (!options.where) { options.where = {}}
     options.where.dao = this.id
-    return Scheme.search(this.context, options, apolloQueryOptions)
+    return ProposalPlugin.search(this.context, options, apolloQueryOptions)
```

Comment:
We want `Plugin.search(...)` here. We also want to support `proposalPlugins(...)`